app/src/consumer/parser_impl_con.c

```diff
@@ -708,11 +728,18 @@ __Z_INLINE parser_error_t _readEmail(parser_tx_t *v, CborValue *rootItem) {
     // email: an e-mail address associated with the entity (string, optional, max 32 characters, must be a valid e-mail address)
     CborValue emailField;
     CborValue dummy;
+    size_t cbor_email_length;
 
     CHECK_CBOR_ERR(cbor_value_map_find_value(rootItem, "email", &emailField))
     if (!cbor_value_is_valid(&emailField))
         return parser_ok;
 
+    CHECK_PARSER_ERR(cbor_value_get_string_length(&emailField, &cbor_email_length))
+    if (cbor_email_length > ENTITY_METADATA_EMAIL_MAX_CHAR) {
```

Comment:
Are we taking into account zero termination here?
I am worried that the size can match exactly the preallocated buffer size and result in issues

Reply:
can we check this is not a problem here or in other similar fields?

Reply:
I believe that the number of characters is taken from the encoded CBOR blob. In CBOR encoding I don't think the text fields are finished using a null character. The length is given at the begining of the string and then we get as much characters as that was indicated from the blob.

See as an example this blob in cbor.me :
```               a76176016375726c7568747470733a2f2f6d792e656e746974792f75726c646e616d6578335468697320697320736f6d6520746f6f6f6f6f6f6f6f6f6f6f6f6f6f6f206c6f6e6720656e74697479206e616d65202835312965656d61696c6d6d7940656e746974792e6f72676673657269616c01676b657962617365716d795f6b6579626173655f68616e646c656774776974746572716d795f747769747465725f68616e646c65
```
So it should be fine, right?

Reply:
> I believe that the number of characters is taken from the encoded CBOR blob. In CBOR encoding I don't think the text fields are finished using a null character. The length is given at the begining of the string and then we get as much characters as that was indicated from the blob.

Indeed. From the [RFC 7049](https://tools.ietf.org/html/rfc7049#section-2.1):

_Major type 3:  a text string, specifically a string of Unicode
      characters that is encoded as UTF-8 [RFC3629].  The format of this
      type is identical to that of byte strings (major type 2), that is,
      as with major type 2, the length gives the number of bytes. 
      [... trimmed ...}_

> See as an example this blob in cbor.me : a76176016375726c7568747470733a2f2f6d792e656e746974792f75726c646e616d6578335468697320697320736f6d6520746f6f6f6f6f6f6f6f6f6f6f6f6f6f6f206c6f6e6720656e74697479206e616d65202835312965656d61696c6d6d7940656e746974792e6f72676673657269616c01676b657962617365716d795f6b6579626173655f68616e646c656774776974746572716d795f747769747465725f68616e646c65

The value of the `name` field in the above CBOR example is a text string of length 51 and is represented as:

```
0b011_11000 (major type 3, additional information 24 to indicate a one-byte length)
0b0011_0011 (51, the actual length)
51 bytes of binary content
```